compiler-cli/src/fs.rs

```diff
@@ -323,6 +330,21 @@ fn is_gleam_path(path: &Path, dir: impl AsRef<Path>) -> bool {
     )
 }
 
+fn is_gleam_ext(path: &Path, dir: impl AsRef<Path>) -> bool {
```

Comment:
No need to check the path prefix here so we can remove the `dir` argument, and can check the extension directly rather than using a regex. Something like this:

```rust
path.extension() == Some("gleam")
```